paimon-core/src/main/java/org/apache/paimon/CoreOptions.java

```diff
@@ -755,6 +755,15 @@ public class CoreOptions implements Serializable {
                             "Parameter string for the constructor of class #. "
                                     + "Callback class should parse the parameter by itself.");
 
+    public static final ConfigOption<Boolean> PARTITION_ADD_TO_METASTORE =
+            key("partition.add-to-metastore")
```

Comment:
I have a feeling that what is actually being said here is that we will synchronize paimon's partition table into Hive's partition table, and after synchronization, of course, we need to addPartition.

So this key better say create hive partitioned table?